openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -659,3 +662,65 @@ def password_reset_token_validate(request):
         AUDIT_LOG.exception("Invalid password reset confirm token")
 
     return JsonResponse({'is_valid': is_valid})
+
+
+def _check_token_has_required_values(**kwargs):
+    """
+    Helper function to test that token
+    string passed has the required kwargs needed
+    to process token validation.
+    """
+    uidb36 = kwargs.get('uidb36')
+    token = kwargs.get('token')
+
+    if not uidb36 or not token:
+        return False
+    try:
+        _ = base36_to_int(uidb36)
+    except ValueError:
+        return False
+    return True
+
+
+@require_POST
+@ensure_csrf_cookie
+def password_reset_logistration(request, **kwargs):
+    """Reset learner password using passed token and new credentials"""
+
+    reset_status = False
+    uidb36 = kwargs.get('uidb36')
+    token = kwargs.get('token')
+
+    if not _check_token_has_required_values(**kwargs):
```

Comment:
Can we pass the already extracted `uidb36` and `token` to this function instead of passing the kwargs and getting these again?